sql/window.go

```diff
@@ -83,6 +85,29 @@ func (w *Window) String() string {
 	return sb.String()
 }
 
+func (w *Window) PartitionId() (uint64, error) {
```

Comment:
Relatively expensive. If this gets called a lot, should probably cache the result.

Reply:
how? if i'm keeping a cache of strings to int64 shouldn't i just use the strings as the ID?

Reply:
i see what you mean now, fixed